pkg/apiserver/rest/dualwriter.go

```diff
@@ -275,20 +275,18 @@ func Compare(storageObj, legacyObj runtime.Object) bool {
 	if storageObj == nil || legacyObj == nil {
 		return storageObj == nil && legacyObj == nil
 	}
-	return bytes.Equal(removeMeta(storageObj), removeMeta(legacyObj))
+	return bytes.Equal(extractSpec(storageObj), extractSpec(legacyObj))
```

Comment:
Should we add a test case to [TestCompare](https://github.com/grafana/grafana/blob/main/pkg/apiserver/rest/dualwriter_test.go#L66) to prevent any regressions?

Reply:
added!